default.nix

```diff
@@ -57,12 +63,12 @@ let
   # fetched binaries are the "standard" builds that people test. This should be
   # fast as it mostly fetches Hydra (CI) caches without building much.
   cardano-node = import
-    (pkgs.fetchgit {
+    (builtins.fetchGit {
       url = "https://github.com/input-output-hk/cardano-node";
-      rev = "1.35.3";
-      sha256 = "020fwimsm24yblr1fmnwx240wj8r3x715p89cpjgnnd8axwf32p0";
+      ref = "purity";
```

Comment:
what is purity?

Reply:
@dermetfan Do we need this `purity` branch? What's the rationale/consequence of not having this? We are asserting that we are using `cardano-node` `1.35.3` in other locations and this branch seems to be on the `master` branch of node.

Reply:
Changed it to point to a new `purity-backport-1.35.4` branch.